src/transformers/models/jetmoe/modeling_jetmoe.py

```diff
@@ -0,0 +1,1731 @@
+    def __init__(self, num_experts: int, input_size: int, output_size: int) -> None:
+        """
+        Initialize the JetMoeParallelExperts module.
+
+        Args:
+            num_experts (int):
+                Number of experts.
+            input_size (int):
+                Size of the input.
+            output_size (int):
+                Size of the output.
+        """
+        super().__init__()
+        self.weight = nn.Parameter(torch.empty(num_experts, output_size, input_size))
+        self.num_experts = num_experts
+        self.input_size = input_size
+        self.output_size = output_size
+
+    def extra_repr(self):
+        return "num_experts={}, input_size={}, output_size={}".format(
+            self.num_experts, self.input_size, self.output_size
+        )
+
+    def forward(self, inputs, expert_size):
+        """
+        Forward pass of the JetMoeParallelExperts module.
+
+        Args:
+            inputs (Tensor):
+                Input tensor.
+            expert_size:
+                Expert size information.
+
+        Returns:
+            Tensor: Output tensor.
+        """
+        input_list = inputs.split(expert_size, dim=0)
+        output_list = []
+        for i in range(self.num_experts):
+            output_list.append(F.linear(input_list[i], self.weight[i]))
+        results = torch.cat(output_list, dim=0)
+        return results
+
+
+class JetMoeTopKGating(nn.Module):
+    def __init__(self, input_size: int, num_experts: int, top_k: int):
+        """
+        Initialize the top-k gating mechanism.
+
+        Args:
+            input_size (`int`):
+                Size of the input.
+            num_experts (`int`):
+                Number of experts.
+            top_k (`int`):
+                Number of top experts to select.
+        """
+        super().__init__()
+
+        self.num_experts = num_experts
+        self.input_size = input_size
+        self.top_k = top_k
+
+        self.layer = nn.Linear(input_size, num_experts, bias=False)
+
+    def extra_repr(self):
+        """
+        Return extra representation string for the module.
+        """
+        return "k={}, num_experts={}".format(self.top_k, self.num_experts)
+
+    def compute_topo(self, top_k_indices, top_k_gates):
+        zeros = torch.zeros(
+            [top_k_gates.size(0), self.num_experts], dtype=top_k_gates.dtype, device=top_k_gates.device
+        )
+        gates = zeros.scatter(1, top_k_indices, 1)
+        expert_size = gates.long().sum(0)
+        top_k_gates = top_k_gates.flatten()
+        top_k_experts = top_k_indices.flatten()
+        _, index_sorted_experts = top_k_experts.sort(0)
+        batch_index = index_sorted_experts.div(self.top_k, rounding_mode="trunc")
+        batch_gates = top_k_gates[index_sorted_experts]
+        expert_size = expert_size.tolist()
+
+        return index_sorted_experts, batch_index, batch_gates, expert_size
+
+    def forward(self, x, return_topo=False):
```

Comment:
```suggestion
    def forward(self, x):
```
It seems `return_topo` is always hard-coded to `True` below. Maybe let's just stick to `True` ?